### PR TITLE
Fix InteractiveQuestion loader show no question before loaded

### PR DESCRIPTION
**Pull request by WiNloSt**
### Description

Reported on [Slack](https://metaboat.slack.com/archives/C063Q3F1HPF/p1714723290124359)

The problem occurred because how we set `loading` state to false. We do that after we wait for `initializeQBRaw` action to be finished which isn't enough. A lot of downstream functions aren't even `await`ed. So we can only rely on `queryResults` to be there.

### How to verify

Use customer zero with this the SDK built with this PR, when going to the product detail page (clicking on a product on the product list page). The first bar chart shouldn't show `no question` error before it's actually loaded.

#### Before
https://github.com/metabase/metabase/assets/1937582/9226cbed-d78a-4a9d-903c-923d9fd21a3a

#### After
https://github.com/metabase/metabase/assets/1937582/b5c0223f-0074-47b2-9f82-8fc7af193e6b

### Checklist

- [ ] Tests have been added/updated to cover changes in this PR

### CODE REVIEW

**Comment by heypoom** [comment]
Verified to work on my end - awesome!

https://github.com/metabase/metabase/assets/4714175/7016a12b-ac31-4ea2-9468-3fbf16ef7da9

File: enterprise/frontend/src/embedding-sdk/components/public/InteractiveQuestion/InteractiveQuestion.tsx [hunk 1]
```diff
@@ -81,7 +81,6 @@ export const _InteractiveQuestion = ({
       await dispatch(initializeQBRaw(location, params));
     } catch (e) {
       console.error(`Failed to get question`, e);
-    } finally {
```
Comment (WiNloSt) [inline enterprise/frontend/src/embedding-sdk/components/public/InteractiveQuestion/InteractiveQuestion.tsx:84]: This is the fix, `finally` will run even when `initializeQBRaw` dosen't throw any error. This isn't enough to say everything is already loaded. The only thing that's guaranteed to be loaded by the time we this action returns is the question.

File: enterprise/frontend/src/embedding-sdk/components/public/InteractiveQuestion/InteractiveQuestion.unit.spec.tsx [hunk 2]
```diff
@@ -83,9 +83,7 @@ describe("InteractiveQuestion", () => {
     expect(screen.getByTestId("loading-spinner")).toBeInTheDocument();
   });
 
-  // TODO [Oisin]: fix failing test in "Fix Interactive Question" PR
-  // eslint-disable-next-line jest/no-disabled-tests
-  it.skip("should render when question is valid", async () => {
```
Comment (WiNloSt) [inline enterprise/frontend/src/embedding-sdk/components/public/InteractiveQuestion/InteractiveQuestion.unit.spec.tsx:88]: This PR also fix this test. I try running with the previous version and it's still fail

File: enterprise/frontend/src/embedding-sdk/components/public/InteractiveQuestion/InteractiveQuestion.unit.spec.tsx [hunk 3]
```diff
@@ -100,6 +98,15 @@ describe("InteractiveQuestion", () => {
     ).toBeInTheDocument();
   });
 
+  it("should not render an error if a question isn't found before the question loaded", async () => {
```
Comment (WiNloSt) [inline enterprise/frontend/src/embedding-sdk/components/public/InteractiveQuestion/InteractiveQuestion.unit.spec.tsx:101]: You can verify this test by running this against the previous version of the code and it should fail.

**Review by heypoom — APPROVED** [review]
Verified to work per the attached video above. Changes looks good. Thanks!

**Review by oisincoveney — APPROVED** [review]

**Pull request merged**